### PR TITLE
Fix multiple issues with user defined data

### PR DESCRIPTION
**Pull request by classicsong**
This PR fix several bugs and gives some enhancement of supporting user defined data
- [x] [BUG] #85 , Force user to provide dataset name when using udd or raw_udd. (This avoid set 'FB15k' as dataset name of user defined data.
- [x] [Enhance], Add check for udd input
- [x] [Doc] #99 , better doc about udd
- [x] [Enhance] #97 , allow users to specify the delimiter

### CODE REVIEW

**Comment by schelv** [comment]
Hi, 
I have two suggestions/questions that would make it easier to train using a User-Defined knowledge graph.

1. 
from the new documentation:
> Here we assume the both the entities ids and relation ids start from 0 and should be contineous

Is it possible to drop this assumption?
I have a dataset that already has a mapping of entity names to ids, and where the ids are not continuous.
Otherwise I would have to create another mapping from the old id to the new id..? .. or use the raw_udd option and let the program do this automatically.
Both options do not seem practical.

2.
Allow to directly specify the relevant column indices of the input files (e.g. `triplets_column_indices=[1, 0, 2]`):
Now you have to specify the format `htr, rht,` etc. which is converted internally with `_parse_srd_format` to `[0,1,2], [1,0,2],` etc.
The advantage of specifying this directly is that it would also allow input files with unused columns (such as qualifiers or sources). 

It would also be great if this is possible for the id mapping files.
The dataset that I want to use has `property_id, en_label, en_description` as the columns.
This cannot be loaded with the code from this pull request, since the label and id are in the wrong order, and there is an unused column.
Specifying something like `relations_map_column_indices=[1,0]` would be very convenient.


Thanks for the awesome repository!

**Comment by classicsong** [comment]
For the first point. Currently we require the node id to be continuous that the trained embedding will not contains untrained data. This assumption can be released if we all used defined dataloader. Because user needs to provide what is the max node ID.

Currently I suggest you using raw_udd

File: python/fake_data/raw_udd/test.tsv [hunk 1]
```diff
@@ -0,0 +1,3 @@
+0|1|9
```
Comment (zheng-da) [inline python/fake_data/raw_udd/test.tsv:1]: if the fake data is used for testing, maybe move them to the test folder.

File: python/dglke/tests/test_dataset.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+        files = gen_udd_files(extra_test=extra_test)
+        with self.assertRaises(AssertionError):
+            dataset = get_dataset('./',
+                                  'udd_test',
+                                  'udd_hrt',
+                                  '\t',
+                                  files)
+        cleanup(files)
+
+    def test_udd_error_nids(self):
```
Comment (zheng-da) [inline python/dglke/tests/test_dataset.py:231]: it has the same name as the method above?

File: python/dglke/tests/test_dataset.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        f.write('A\t0\n')
+
+    return [emap, rmap, train_f, valid_f, test_f]
+
+def cleanup(files):
+    for f in files:
+        os.remove(f)
+
+class TestUDDDataset(unittest.TestCase):
+    def test_udd_noint_triplets(self):
```
Comment (zheng-da) [inline python/dglke/tests/test_dataset.py:67]: could you write comments to describe what error the test tries to check?

File: tests/scripts/task_kg_test.sh [hunk 4]
```diff
@@ -53,7 +59,71 @@ if [ "$2" == "cpu" ]; then
     dglke_eval --model_name DistMult --dataset FB15k --hidden_dim 100 \
         --gamma 500.0 --batch_size 16 --eval_percent 0.01 --model_path ckpts/DistMult_FB15k_0/ \
         --data_path /data/kg || fail "eval DistMult on $2"
+
+    rm -fr ckpts/
+    # udd test
```
Comment (zheng-da) [inline tests/scripts/task_kg_test.sh:64]: is it right that they try to test different delimiters?

File: python/fake_data/raw_udd/test.tsv (hunk 1 above)
Comment (classicsong) [reply python/fake_data/raw_udd/test.tsv:1]: OK

File: python/dglke/tests/test_dataset.py (hunk 3 above)
Comment (classicsong) [reply python/dglke/tests/test_dataset.py:67]: OK

File: python/dglke/tests/test_dataset.py (hunk 2 above)
Comment (classicsong) [reply python/dglke/tests/test_dataset.py:231]: It should be test_udd_error_rids.

File: tests/scripts/task_kg_test.sh (hunk 4 above)
Comment (classicsong) [reply tests/scripts/task_kg_test.sh:64]: Yes, we test 4 different commonly used delimiters

**Pull request merged**